api/placer.py

```diff
@@ -0,0 +1,291 @@
+import bson
+import copy
+import datetime
+import dateutil
+import os
+import pymongo
+import zipfile
+
+from . import base
+from . import config
+from . import files
+from . import rules
+from . import tempdir as tempfile
+from . import util
+from . import validators
+from .dao import reaperutil, APIStorageException
+
+
+class Placer(object):
+    """
+    Interface for a placer, which knows how to process files and place them where they belong - on disk and database.
+    """
+
+    def __init__(self, container_type, container, id, metadata, timestamp):
+        self.container_type = container_type
```

Comment:
So far, I've only been using `container_type, container, id` for a single target, and leaving it blank when there are multiple hierarchical interactions (such as the PackfilePlacer).

I think this needs to be addressed across the entire codebase: we don't have a consistent way to describe a container (which may or may not exist in memory at the moment), and when we establish that perhaps a placer takes an array of targets, or something.